packages/design-system/src/components/FilterTag/FilterTag.tsx

```diff
@@ -0,0 +1,104 @@
+import { ClearIcon } from '../ClearIcon';
+import React from 'react';
+import classNames from 'classnames';
+
+const actionableKeys = ['Enter', 'Backspace', 'Delete'];
+const idFromValue = (value: string) => `${value}-tag`;
+
+export interface FilterTagProps {
+  /**
+   * Id for filter tag button. If not provided, a unique id will be automatically generated and used.
+   */
+  id?: string;
+  /**
+   * Class to be applied to the outer `<div>` that contains filter tag button.
+   */
+  className?: string;
+  /**
+   * Text for the filter tag
+   */
+  label: string;
+  /**  
+   *
+   Value for filter tag 
+   */
+  value: string;
+  /**
+   *  For screenreaders, text to read for removal
+   */
+  removeText: string;
+  /**
+   * When false, filter tag is not removable
+   */
+  readOnly?: boolean;
```

Comment:
I'm wondering about if we change `readOnly` to something like `notDismissible` or something that lets folks know they can't remove or dismiss it

Reply:
Or if the `removeText` prop requirement becomes conditional, maybe the name could even change to something related to that prop, like `removeable`. Or this could actually be removed entirely, and it becomes removable if `onClear` is defined. And maybe `onClear` gets renamed to `onRemove`? These names don't need to match pet-shared in order for us to feel comfortable migrating away from our implementation 🙂

Reply:
or change `onClear` to  `onClick?

Reply:
@line47 @nichia What do you think about removing the readOnly prop, renaming `onClear` to `onDelete`, and only having the X icon show if `onDelete` function is passed on? That is the way this filter chip behaves: https://material-ui.com/components/chips/

Can anyone think of a use case where we would need this to be removable, but a callback onDelete function wouldn't be necessary? I can't..

Reply:
Yes, that would work well.